### PR TITLE
Adds a cronjob-like mechanism

### PR DESCRIPTION
**Pull request by benvia**
Based on the issue #214 I propose a solution to use DB_DUMP_BEGIN as a base time to run like a cronjob.

If the DB_DUMP_BEGIN is specified in hh:mm:ss format, then the system uses it as a base time with DB_DUMP_FREQ as the recursion.

I'm open to comments about this solutions.

### CODE REVIEW

**Comment by tiredofit** [comment]
Hi @benvia - Thanks for the PR and apologies on the delay. I am in the midst of preparing a major release for db-backup and reviewing PRs. If I am to understand correctly entering in a HH:MM:SS format could take the place of the existing HHMM format which is not persisting after a container restart?

I am open is this will resolve the issue that this originally came up, could I request some examples and if I respond that this makes sense to go in perhaps some explanation in the README.md file?  I would likely be refactoring your code to fit with the new scheduling components on my end but the main bulk of your work would be there in inspiration. I am looking to release this month after it simmers for a bit and I understand if there are any hidden bugs I haven't caught.

**Comment by benvia** [comment]
Hi @tiredofit,

Thank you for your feedback.

If you put a HH:mm:ss value for DB_DUMP_BEGIN parameter then the first backup will be executed at this time. If this time is in the past then the function will calculate the next execution based on time of DB_DUMP_BEGIN and DB_DUMP_FREQ.

This solution is partially answering the issue #214 as the code is not taking care of the date but only the time. I could take a look to extend the code to support date also.

Example 1:
Current time : 07:04:01
DB_DUMP_BEGIN : 06:00:00
DB_DUMP_FREQ : 10
=> Next backup 07:10:00

Example 2:
Current time : 07:04:01
DB_DUMP_BEGIN : 10:25:00
DB_DUMP_FRE : 10
=> Next backup 10:25:00

**Comment by benvia** [comment]
I've modify the code to support a full date and time value. Now the DB_DUMP_BEGIN must be a full date and time value in form YYYY-MM-DD HH:mm:ss (ex: 2023-10-03 18:05:30)

**Comment by tiredofit** [comment]
Excellent! This will work out and should solve the original issue and add another format of scheduling.

File: install/etc/services.available/10-db-backup/run [hunk 1]
```diff
@@ -27,15 +27,18 @@ else
         print_debug "DB_DUMP_BEGIN is an integer."
         waittime=$(( ${BASH_REMATCH[1]} * 60 ))
```
Comment (tiredofit) [inline install/etc/services.available/10-db-backup/run:28]: 28-29 are off - they should actually be: 

```
        target_time=$(date --date="${today}${DB_DUMP_BEGIN}" +"%s")
        if [[ "$target_time" < "$current_time" ]]; then
            target_time=$(($target_time + 24*60*60))
        fi
        waittime=$(($target_time - $current_time))
```

Comment (benvia) [reply install/etc/services.available/10-db-backup/run:28]: Thank you.

**Comment by tiredofit** [comment]
Alright! We're merged. I've reworked in upcoming 4.x series your work, and it will be tagged as part of the new release shortly.

**Pull request merged**